test/rpc/test_rpc_traces.py

```diff
@@ -0,0 +1,307 @@
+    """Test simulate_transaction with declare v2 transaction"""
+    contract_class, _, compiled_class_hash = load_cairo1_contract()
+
+    nonce = get_nonce(PREDEPLOYED_ACCOUNT_ADDRESS)
+
+    tx_hash = calculate_declare_transaction_hash(
+        contract_class=contract_class,
+        compiled_class_hash=compiled_class_hash,
+        chain_id=StarknetChainId.TESTNET.value,
+        sender_address=int(PREDEPLOYED_ACCOUNT_ADDRESS, 16),
+        max_fee=0,
+        version=SUPPORTED_RPC_DECLARE_TX_VERSION,
+        nonce=nonce,
+    )
+
+    signature = _get_signature(tx_hash, PREDEPLOYED_ACCOUNT_PRIVATE_KEY)
+
+    declare_transaction = RpcBroadcastedDeclareTxnV2(
+        contract_class=rpc_contract_class(contract_class),
+        sender_address=PREDEPLOYED_ACCOUNT_ADDRESS,
+        compiled_class_hash=rpc_felt(compiled_class_hash),
+        type="DECLARE",
+        version=rpc_felt(SUPPORTED_RPC_DECLARE_TX_VERSION),
+        nonce=rpc_felt(nonce),
+        max_fee=rpc_felt(0),
+        signature=list(map(rpc_felt, signature)),
+    )
+
+    response = rpc_call_background_devnet(
+        "starknet_simulateTransaction",
+        {
+            "block_id": "latest",
+            "transaction": [declare_transaction],
+            "simulation_flags": simulation_flags,
+        },
+    )
+
+    if not simulation_flags:
+        assert (
+            response["result"][0]["fee_estimation"][0]["overall_fee"] == "0x1d2c764500"
+        )
+        assert response["result"][0]["transaction_trace"][0]["validate_invocation"][
+            "contract_address"
+        ] == rpc_felt(PREDEPLOYED_ACCOUNT_ADDRESS)
+    else:
+        assert (
+            response["result"][0]["fee_estimation"][0]["overall_fee"] == "0x1d208a8300"
+        )
+        assert (
+            response["result"][0]["transaction_trace"][0]["validate_invocation"] is None
+        )
+
+    assert (
+        response["result"][0]["transaction_trace"][0]["fee_transfer_invocation"] is None
+    )
+
+
+@pytest.mark.usefixtures("run_devnet_in_background")
+@pytest.mark.parametrize(
+    "run_devnet_in_background, simulation_flags",
+    [
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            [],
+        ),
+        ([*PREDEPLOY_ACCOUNT_CLI_ARGS], [SimulationFlag.SKIP_VALIDATE.name]),
+    ],
+    indirect=["run_devnet_in_background"],
+)
+def test_simulate_transaction_deploy_account(simulation_flags, deploy_account_details):
+    """Test simulate_transaction with deploy account transaction"""
+    deploy_account_tx, _ = prepare_deploy_account_tx(**deploy_account_details)
+    rpc_deploy_account_tx = rpc_deploy_account_from_gateway(deploy_account_tx)
+
+    response = rpc_call_background_devnet(
+        "starknet_simulateTransaction",
+        {
+            "block_id": "latest",
+            "transaction": [rpc_deploy_account_tx],
+            "simulation_flags": simulation_flags,
+        },
+    )
+
+    if not simulation_flags:
```

Comment:
I don't like these `if`s in test cases, especially with only two parametrized subcases. How about this:
- Instead of hardcoding the `overall_fee` values, the same test case could first run it with simulation_flag, then without, and then assert that both values are > 0 and that with skipped simulation the fee is smaller. This assertion could even be done with a common utility function.
- The current test parametrization is complicated, with this suggestion, there would be no parametrization, and we could use the `devnet_in_background` decorator.

This applies to all four test cases in this file.

Reply:
Let's discuss it later on a call.